shuttle/src/runtime/task/mod.rs

```diff
@@ -146,11 +149,95 @@ where
     }
 }
 
+/// A task signature is an identifier that is intended to be *mostly* stable across executions
+/// and allow for categorization of tasks according to how they were created. It provides three
+/// levels of granularity: static (compile-time) spawn location, dynamic (run-time) context where
+/// that spawn location was reached. The static spawn location and signature are each represented
+/// by a u64 so that the details of how they are computed can be non-breaking changes in the future.
```

Comment:
This is a non-care. None of this is pub so all changes are non-breaking. The reason we have `u64`s is because we want to do efficient comparisons.